source/agora/common/Serializer.d

```diff
@@ -18,6 +18,43 @@ import agora.common.Data;
 /// Type of delegate SerializeDg
 public alias SerializeDg = void delegate(scope const(ubyte)[]) @safe;
 
+/// Default serializer implementation
+public mixin template DefaultSerializer ()
+{
+    static assert(is(typeof(this) == struct),
+        "`DefaultSerializer` needs to be mixed in a `struct` context");
+
+    /***************************************************************************
+
+        Serialize all members of this `struct`, in the order they appear.
+
+        Params:
+            dg = serialize function accumulator
+
+    ***************************************************************************/
+
+    public void serialize (scope SerializeDg dg) const @safe
+    {
+        foreach (ref entry; this.tupleof)
```

Comment:
`const ref`?

Reply:
It's `const` by default because it's part of `this`, but yeah